game.cpp

```diff
@@ -69,6 +72,8 @@ Game::~Game()
 	delete events;
 	ppUnits = 0;
 	maxppunits = 0;
+  	// Return the global array to it's original state. (needed for unit tests)
+  	FactionTypes->clear();
```

Comment:
When I was testing, when I had multiple tests which set up independent games, I discovered this bug.
It's only a problem within the testing framework where you might create multiple game objects to run small tests across various files and each one would add the types configured for that test to the types array again.